core/src/message/context.rs

```diff
@@ -224,31 +186,47 @@ impl ContextStore {
 }
 
 /// Context of currently processing incoming message.
-#[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
+#[derive(Debug)]
 pub struct MessageContext {
     kind: DispatchKind,
     current: IncomingMessage,
     outcome: ContextOutcome,
     store: ContextStore,
     settings: ContextSettings,
+    outgoing_bytes_counter: u32,
 }
 
 impl MessageContext {
-    /// Create new MessageContext with given ContextSettings.
+    /// Create new message context.
+    /// Returns `None` if outgoing messages bytes limit exceeded.
     pub fn new(
         dispatch: IncomingDispatch,
         program_id: ProgramId,
         settings: ContextSettings,
-    ) -> Self {
+    ) -> Option<Self> {
```

Comment:
NOTE: now returns Option<Self>. May be better return an error, but I do not see the difference